### PR TITLE
Convert the OrgMembers store into a basestore

### PR DESCRIPTION
**Pull request by asdine**
Fixes https://github.com/sourcegraph/sourcegraph/issues/15850

### CODE REVIEW

File: internal/db/users.go [hunk 1]
```diff
@@ -341,7 +341,7 @@ func (u *UserStore) create(ctx context.Context, info NewUser) (newUser *types.Us
 		for _, err := range errs {
 			log15.Warn(err.Error())
 		}
-		if err := OrgMembers.CreateMembershipInOrgsForAllUsers(ctx, u, orgs); err != nil {
+		if err := OrgMembers.With(u).CreateMembershipInOrgsForAllUsers(ctx, orgs); err != nil {
```
Comment (ryanslade) [inline internal/db/users.go:344]: I think we need to pass u in here so that we can reuse the transaction

**Review by ryanslade — CHANGES_REQUESTED** [review]
I think you need to handle passing in a transaction to `CreateMembershipInOrgsForAllUsers`

File: internal/db/users.go (hunk 1 above)
Comment (asdine) [reply internal/db/users.go:344]: Isn't that what `with` is for? Calling OrgMembers with `With(u)` will reuse the same database handle and achieve the same result as before right?

Comment (ryanslade) [reply internal/db/users.go:344]: Ah, yes.. 🤦

**Review by ryanslade — APPROVED** [review]

**Pull request merged**